api/authenticator.py

```diff
@@ -1277,9 +1277,12 @@ def create_authentication_headers(self):
         return headers
 
 
-class AuthenticationProvider(OPDSAuthenticationFlow):
+class AuthenticationProvider(OPDSAuthenticationFlow, ABC):
     """Handle a specific patron authentication scheme."""
```

Comment:
`AuthenticationProvider` was conceptually an abstract class without actually saying so. By extending `ABC`, we're making it a hard requirement to fill in the methods marked as abstract; it's no longer possible to instantiate the class directly. This helped massively in finding all of the places in the codebase that needed updating when a new method was added.

Reply:
IIRC `ABC` is new in Python 3.x, so this code was written before it was a thing. Makes sense to start using it here.